Sources/Tracking/Background Queue/Type/QueueTaskMetadata.swift

```diff
@@ -3,13 +3,22 @@ import Foundation
 /// Pointer to full queue task in persistent storage.
 /// This data structure is meant to be as small as possible with the
 /// ability to hold all queue task metadata in memory at runtime.
-public struct QueueTaskMetadata: Codable, Equatable {
+public struct QueueTaskMetadata: Codable, Equatable, AutoLenses {
     let taskPersistedId: String
     let taskType: String
+    /// Groups that this task belongs to where this task is the parent of the group.
+    /// Tasks can be the parent of 0+ groups
```

Comment:
do you have an idea of a case where we'd want something to parent multiple groups? I don't really know that it's a use-case we need to support 🤔

Reply:
No, I can't think of a scenario where you would need to today. 

In the future if we ever decided to change from `groupStart: String` to `groupStart: [String]` it would require us writing some code to migrate the queue saved in persistent storage on the device. 

I would like to avoid having to write this migration code but we probably need to write it at some time in the future anyways as we will more then likely have a breaking change at some point. 

So, I changed the to only allowing 0 or 1 `groupStart` and if we need to change it in the future, we will migrate the storage to allow it.